shark_turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -31,11 +35,16 @@ def partition_strided_operators(trace: CapturedTrace, constraints: list[Constrai
 
     def has_strided_access(node: fx.Node) -> bool:
         """
-        Checks for writes on 2d tensors with strided access on a single dimension.
+        Checks for writes on 2d tensors with strided access on a single dimension that
+        read more than a single element.
         """
         custom = get_custom(node)
         if isinstance(custom, Write) and len(custom.type.symbolic_shape) == 2:
             strides = [simplify_index(custom.index[dim]).stride for dim in custom.index]
+            elements_per_thread = [
+                simplify_index(custom.index[dim]).size for dim in custom.index
+            ]
+            strides = [x for x, y in zip(strides, elements_per_thread) if y > 1]
```

Comment:
I thought we only care for fastest dimensions no?

Reply:
I guess there wouldn't/shouldn't be a case where fastest dim's elem_per_thread == 1, while the a non fastest > 1. => shouldn't matter!

Reply:
So with this patch for a tensor of [M, N] the stride will be [N, 1]. Now lets say we are reading 4 elements along the N dimension. So our access patterns for N will be [offset, read 4 elements, stride 1] and our access pattern for M will be [offset, read 1 element, stride N]. Without this check, this pattern would get applied to this tensor which we don't want. So that's why its not enough to just check for strided access, we also need to know how many elements are being loaded. In cases where we are reading 4 elements from N, our access pattern for N will be [offset, read 1 element, stride 1] and for M will be [offset, read 4 elements, stride N] and then we do want this pattern to kick in.

Reply:
makes sense, thanks!